File: 01_numbers/python/variables.py
if __name__ == '__main__':

    x = 1
    print(x)
    # bind x again shadows old one from above
    x = 'i'
    print(x)

    # declare, init
    something = None
    x = 5
    print("x, something", x, something)
    something = x * 5
    print("x, something", x, something)

    # mutability
    y = 0
    y = y * 2 + x
    print(y)

    BLAH = 42  # only const by convention
    y *= BLAH

File: 01_numbers/rust/variables.rs
fn main() {

    let x = 1;
    println!("x: {}", x);
    // binds x again, shadowing the old one from above
    let x = 'i';
    println!("x: {}", x);

    // declare, initialize
    let something;
    let x = 5;
    println!("x, something: {}, {}", x, something);
    something = x * 5;
    println!("x, something: {}, {}", x, something);

    // Mutability
    let y = 0;
    y = y * 2 + x;
    dbg!(y);

    const blah: i32 = 42;
    y *= blah;
    dbg!(y);
}
